addons/udes_stock/models/stock_picking_batch.py

```diff
@@ -453,11 +453,112 @@ def drop_off_picked(self, continue_batch, location_barcode):
 
                 # at this point pick_todo should contain only mls done
                 pick_todo.update_picking(validate=True)
+
         if not continue_batch:
             self.unassign()
 
         return self
 
+    @api.model
+    def get_drop_off_instructions(self, criterion):
+        """
+        Returns a string indicating instructions about what the
+        user has to scan before requesting the move lines for
+        drop off.
+
+        Raises an error if the instruction method for the
+        specified criterion does not exist.
+        """
+        func = getattr(self, '_get_drop_off_instructions_' + criterion, None)
+
+        if not func:
+            raise ValidationError(
+                _("An unexpected drop off criterion is currently configured")
+                + ": '%s'" % criterion if criterion else "")
+
+        return func()
+
+    def get_next_drop_off(self, item_identity):
+        """
+        Based on the criteria specified for the batch picking type,
+        determines what move lines should be dropped (refer to the
+        batch::drop API specs for the format of the returned value).
+
+        Expects an `in_progress` singleton.
+
+        Raises an error in case:
+         - not all pickings of the batch have the same picking type;
+         - unknown or invalid (e.g. 'all') drop off criterion.
+        """
+        self.ensure_one()
+        assert self.state == 'in_progress', \
+               "Batch must be in progress to be dropped off"
+
+        all_mls_to_drop = self._get_move_lines_to_drop_off()
+
+        if not len(all_mls_to_drop):
+            return {'last': True,
```

Comment:
Shouldn't this be an error? Does not make sense to have item_identity and no move lines to drop off.

Reply:
That's what the API says;l we tried to make the API robust - you never know what can happen to the system once a request is issued, so, if that happens, you have a specified behaviour.